client/src/views/Home/index.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState, useEffect } from "react";
+import MovieItem from "./MovieItems";
+import SearchBar from "../../components/SearchBar";
+import HomeText from "../../components/HomeText";
+import SearchError from "../../components/SearchError";
+import Loader from "../../components/Loader";
+import Illustration from "../../images/2.Illustrations/illustration-empty-state.png";
+import "./styles.scss";
+import axios from "axios";
+
+const Home = ({ favorites }) => {
+  const [searchText, setSearchText] = useState("");
+  const [moviesList, setMoviesList] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [searchError, setSearchError] = useState(false);
+
+  useEffect(() => {
```

Comment:
This `useEffect` is doing too many things at the same time: 
- Requesting to the API;
- Get loading and error state;
- Debounce;
- Listening to a keydown event;

Consider breaking it down in the smallest pieces, which will do only one thing. It will improve the readability, testability and maintainability of the function.

Reply:
Hi Danilo! 😄 

Would you recommend that I implement multiple `useEffect` functions, each one responsible for less effects, instead of a single one responsible for many effects?

Reply:
Hey
Not really, I'd recommend creating some kind of abstraction, for example:
```js
const searchQueryDebounced = useDebounce(searchQuery)
useEffect(() => { 
 setLoading(true)
 const movies = getMovies(searchQueryDebounced)
 // etc
}, [])
```

Reply:
Abstractions created for the effects in the home and single movie views